Common/example_utils.hpp

```diff
@@ -244,6 +244,42 @@ void multiply_matrices(T        alpha,
     }
 }
 
+/// \brief Prints an {1,2,3}-dimensional array. The last dimension (fastest-index) specified in
+/// \p n will be printed horizontally.
+template<class Tdata, class Tsize>
+void print_nd_data(const std::vector<Tdata> data,
+                   std::vector<Tsize>       np,
+                   const int                column_width     = 4,
+                   const bool               reverse_indexing = false)
```

Comment:
I think this would be easier to understand and use if it was either renamed to `row_major` (since the deffault looks like column-major),  or if this was replaced by an enum that spelled out exactly what the caller wanted.

Reply:
Hm yes, this was added to support printing data with either layout. IINM the default is row-major (`np[0]` is assumed to be the size of the outermost dimension), and when reversing the ordering of the dimension sizes we assume a column-major layout. Changed this parameter to `column_major`.